libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyUtils.java

```diff
@@ -106,11 +110,11 @@ public static Policy parseEncodedPolicyIfExists(
                         version
                     );
                 }
-            } catch (Exception ex) {
-                logger.warn(
-                    Strings.format("Found a policy patch with invalid content. The patch will not be applied. Layer [%s]", layerName),
-                    ex
-                );
+            } catch (PolicyParserException e) {
+                // This is more specific and informative than a generalized IllegalStateException
+                throw e;
+            } catch (IOException | RuntimeException e) {
```

Comment:
Is the RuntimeException necessary to catch? Could we just convert IOException to UncheckedIOException and leave the rest to propagate on their own? There is something to be said for wrapping any exception, though (not just these) since it includes the layer name? Does the PolicyParserException include that?

Reply:
Why not be very clear and convert the old log message (that happened for all Exception) to a runtime exception? The cause will still be included, so the PolicyParserException would still be visible.

Reply:
That's what I did. The parts of the old log message I omitted are no longer true.

Reply:
Are you saying you don't want me to peel off `PolicyParseException`? Sure, that's an easy change.

Reply:
I think you're saying you prefer to catch `Exception` instead of `IOException | RuntimeException`? I can change that too.

They're equivalent, but it's a good practice to list the checked exceptions you're catching, so I refactored it.